contribs/src/main/java/com/netflix/conductor/contribs/tasks/json/JsonJqTransform.java

```diff
@@ -58,7 +59,7 @@ public JsonJqTransform(ObjectMapper objectMapper) {
     @Override
     public void start(Workflow workflow, Task task, WorkflowExecutor executor) {
         final Map<String, Object> taskInput = task.getInputData();
-        final Map<String, Object> taskOutput = task.getOutputData();
+        final Map<String, Object> taskOutput = task.getOutputData() != null ? task.getOutputData(): new HashMap<>();
```

Comment:
`getOutputData()` does not return `null`. Can you please explain the need for a null check here?

Reply:
The outputData field is reset to null here: https://github.com/Netflix/conductor/blob/main/core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java#L1642 I guess we could update that line to just clear the map instead of setting it null, but feel there might be other cases that I might neglected if I changed that. So I added the null check for outputData in the system task instead. And that's the only sync system task that need to add the null check

Reply:
It makes sense to clear the `outputData` or set it to an empty `HashMap` during `rerun`. If you look at the [Task](https://github.com/Netflix/conductor/blob/main/common/src/main/java/com/netflix/conductor/common/metadata/tasks/Task.java#L158), the outputData field is defaulted to an empty map. So it's natural for the rest of the code to expect a non-null value from `Task.getOutputData()`. `DoWhile` also [expects a non-null output data](https://github.com/Netflix/conductor/blob/main/core/src/main/java/com/netflix/conductor/core/execution/tasks/DoWhile.java#L63) and would fail if null is returned.

Also, the fewer nulls we introduce, the less chance of an unexpected `NullPointerException`.

Reply:
ya updated it to clear the `outputData` instead of setting to null